### PR TITLE
Fix webcamsOnlyForModerator

### PR DESCRIPTION
**Pull request by pedrobmarin**

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/video-provider/service.js [hunk 1]
```diff
@@ -215,12 +219,47 @@ class VideoService {
     return streams.find(s => s.stream === stream);
   }
 
+  filterModeratorOnly(streams) {
+    const me = Users.findOne({ userId: Auth.userID });
+    const amIViewer = me.role === ROLE_VIEWER;
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/video-provider/service.js]: can you pass it on find query?

Comment (pedrobmarin) [reply bigbluebutton-html5/imports/ui/components/video-provider/service.js]: the role?

Comment (Tainan404) [reply bigbluebutton-html5/imports/ui/components/video-provider/service.js]: Yes

**Comment by daminebenq** [comment]
tested and working perfectly

**Comment by Tainan404** [comment]
Something isn't allowing users join to client if another user is sharing the webcam in the meeting (I wasn't able to reproduce in the main branch)

Peek 2020-04-01 14-16

The logs that are shown in the gif are below:
```
Uncaught TypeError: Cannot read property 'offsetHeight' of null
    at WebcamDraggable.render (component.jsx:444)
    at finishClassComponent (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:26588)
    at updateClassComponent (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:26541)
    at beginWork$1 (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:28304)
    at HTMLUnknownElement.callCallback (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8454)
    at Object.invokeGuardedCallbackDev (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8503)
    at invokeGuardedCallback (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8558)
    at beginWork$$1 (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33898)
    at performUnitOfWork (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32813)
    at workLoopSync (modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32789)
```

```
react_devtools_backend.js:6 The above error occurred in the <WebcamDraggable> component:
    in WebcamDraggable (created by Context.Consumer)
    in Unknown (created by Media)
    in div (created by Media)
    in Media (created by MediaContainer)
    in MediaContainer (created by InjectIntl(MediaContainer))
    in InjectIntl(MediaContainer) (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent (created by ModalMounterWrapper)
    in ModalMounterWrapper (created by Context.Consumer)
    in Unknown
    in section (created by bound renderMedia)
    in bound renderMedia (created by Context.Consumer)
    in Unknown (created by App)
    in ContextProvider (created by App)
    in div (created by App)
    in section (created by App)
    in main (created by App)
    in App (created by InjectIntl(App))
    in InjectIntl(App) (created by AppContainer)
    in AppContainer (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent (created by ModalMounterWrapper)
    in ModalMounterWrapper (created by InjectIntl(ModalMounterWrapper))
    in InjectIntl(ModalMounterWrapper) (created by Base)
    in IntlProvider (created by IntlStartup)
    in IntlStartup (created by Base)
    in Base (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent
    in Subscriptions (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent
    in AuthenticatedHandler
    in JoinHandler

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
r @ react_devtools_backend.js:6
logCapturedError @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:29961
logError @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:29998
update.callback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:31350
callCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21947
commitUpdateEffects @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21985
commitUpdateQueue @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21975
commitLifeCycles @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:30278
commitLayoutEffects @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33462
callCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8454
invokeGuardedCallbackDev @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8503
invokeGuardedCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8558
commitRootImpl @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33200
unstable_runWithPriority @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:36688
runWithPriority$2 @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20267
commitRoot @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33040
finishSyncRender @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32447
performSyncWorkOnRoot @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32425
(anonymous) @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20317
unstable_runWithPriority @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:36688
runWithPriority$2 @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20267
flushSyncCallbackQueueImpl @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20312
flushSyncCallbackQueue @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20300
scheduleUpdateOnFiber @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:31827
enqueueSetState @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:22112
Component.setState @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:4766
(anonymous) @ intl.jsx:141
(anonymous) @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:1234
Promise.then (async)
Promise.then @ promise.js?hash=bcc278416465049d96746a94376f34245ad33b8c:132
(anonymous) @ intl.jsx:139
callCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21947
commitUpdateEffects @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21985
commitUpdateQueue @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21973
commitLifeCycles @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:30253
commitLayoutEffects @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33462
callCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8454
invokeGuardedCallbackDev @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8503
invokeGuardedCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8558
commitRootImpl @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33200
unstable_runWithPriority @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:36688
runWithPriority$2 @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20267
commitRoot @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33040
finishSyncRender @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32447
performSyncWorkOnRoot @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32425
(anonymous) @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20317
unstable_runWithPriority @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:36688
runWithPriority$2 @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20267
flushSyncCallbackQueueImpl @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20312
flushSyncCallbackQueue @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20300
scheduleUpdateOnFiber @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:31827
enqueueForceUpdate @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:22151
Component.forceUpdate @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:4785
(anonymous) @ ReactMeteorData.jsx:79
_compute @ tracker.js:308
_recompute @ tracker.js:324
Tracker._runFlush @ tracker.js:495
onGlobalMessage @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:515
postMessage (async)
setImmediate @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:535
requireFlush @ tracker.js:129
invalidate @ tracker.js:264
changed @ tracker.js:422
(anonymous) @ cursor.js:293
SQp.runTask @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:849
SQp.flush @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:878
SQp.drain @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:887
resumeObservers @ local_collection.js:306
endUpdate @ collection.js:239
store.<computed> @ livedata_connection.js:310
(anonymous) @ livedata_connection.js:1212
_performWrites @ livedata_connection.js:1211
_flushBufferedWrites @ livedata_connection.js:1167
_livedata_data @ livedata_connection.js:1133
onMessage @ livedata_connection.js:1663
(anonymous) @ browser.js:186
forEachCallback @ common.js:30
socket.onmessage @ browser.js:185
REventTarget.dispatchEvent @ sockjs-0.3.4.js:87
SockJS._dispatchMessage @ sockjs-0.3.4.js:1072
SockJS._didMessage @ sockjs-0.3.4.js:1130
that.ws.onmessage @ sockjs-0.3.4.js:1277
react_devtools_backend.js:6 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in IntlStartup (created by Base)
    in Base (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent
    in Subscriptions (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent
    in AuthenticatedHandler
    in JoinHandler
r @ react_devtools_backend.js:6
warningWithoutStack @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8648
warnAboutUpdateOnUnmountedFiberInDEV @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33856
scheduleUpdateOnFiber @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:31797
enqueueSetState @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:22112
Component.setState @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:4766
(anonymous) @ intl.jsx:146
(anonymous) @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:1234
Promise.then (async)
Promise.then @ promise.js?hash=bcc278416465049d96746a94376f34245ad33b8c:132
(anonymous) @ intl.jsx:131
callCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21947
commitUpdateEffects @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21985
commitUpdateQueue @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:21973
commitLifeCycles @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:30253
commitLayoutEffects @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33462
callCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8454
invokeGuardedCallbackDev @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8503
invokeGuardedCallback @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:8558
commitRootImpl @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33200
unstable_runWithPriority @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:36688
runWithPriority$2 @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20267
commitRoot @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:33040
finishSyncRender @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32447
performSyncWorkOnRoot @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:32425
(anonymous) @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20317
unstable_runWithPriority @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:36688
runWithPriority$2 @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20267
flushSyncCallbackQueueImpl @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20312
flushSyncCallbackQueue @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:20300
scheduleUpdateOnFiber @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:31827
enqueueForceUpdate @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:22151
Component.forceUpdate @ modules.js?hash=4398da1f016fedc04776abdc6994ccc5be190483:4785
(anonymous) @ ReactMeteorData.jsx:79
_compute @ tracker.js:308
_recompute @ tracker.js:324
Tracker._runFlush @ tracker.js:495
onGlobalMessage @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:515
postMessage (async)
setImmediate @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:535
requireFlush @ tracker.js:129
invalidate @ tracker.js:264
changed @ tracker.js:422
(anonymous) @ cursor.js:293
SQp.runTask @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:849
SQp.flush @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:878
SQp.drain @ meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:887
resumeObservers @ local_collection.js:306
endUpdate @ collection.js:239
store.<computed> @ livedata_connection.js:310
(anonymous) @ livedata_connection.js:1212
_performWrites @ livedata_connection.js:1211
_flushBufferedWrites @ livedata_connection.js:1167
_livedata_data @ livedata_connection.js:1133
onMessage @ livedata_connection.js:1663
(anonymous) @ browser.js:186
forEachCallback @ common.js:30
socket.onmessage @ browser.js:185
REventTarget.dispatchEvent @ sockjs-0.3.4.js:87
SockJS._dispatchMessage @ sockjs-0.3.4.js:1072
SockJS._didMessage @ sockjs-0.3.4.js:1130
that.ws.onmessage @ sockjs-0.3.4.js:1277
react_devtools_backend.js:6 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in ReactMeteorDataComponent
    in section (created by App)
    in div (created by App)
    in section (created by App)
    in main (created by App)
    in App (created by InjectIntl(App))
    in InjectIntl(App) (created by AppContainer)
    in AppContainer (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent (created by ModalMounterWrapper)
    in ModalMounterWrapper (created by InjectIntl(ModalMounterWrapper))
    in InjectIntl(ModalMounterWrapper) (created by Base)
    in IntlProvider (created by IntlStartup)
    in IntlStartup (created by Base)
    in Base (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent
    in Subscriptions (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent
    in AuthenticatedHandler
    in JoinHandler
```

**Comment by pedrobmarin** [comment]
Is this happening while locking cams for moderator only? If so, could you log the content of this property? https://github.com/bigbluebutton/bigbluebutton/blob/41f19a2cd1bc7dae76cbd805cdc3ddfbf1e6ab18/bigbluebutton-html5/imports/ui/components/media/container.jsx#L126

**Comment by Tainan404** [comment]
That's it stored in the variable
[{"cameraId":"w_qpvfdannts7d_a7dd78b449c506f89820069d7607e30e02f9b6e211d26d4631625b2c6f9377d4","userId":"w_qpvfdannts7d","name":"156"}]"

**Pull request merged**